R/tax_range_time.R

```diff
@@ -0,0 +1,158 @@
+#' Calculate the temporal range of fossil taxa
+#'
+#' A function to calculate the temporal range of fossil taxa from occurrence
+#' data.
+#'
+#' @param occdf \code{dataframe}. A dataframe of fossil occurrences containing
+#' at least three columns: names of taxa, maximum age and minimum age.
+#' These ages should constrain the age range of the fossil occurrence
+#' and are assumed to be in millions of years before present.
+#' @param name \code{character}. The name of the column you wish to be treated
+#' as the input names, e.g. "genus" (default).
+#' @param min_ma \code{character}. The name of the column you wish to be treated
+#' as the minimum limit of the age range, e.g. "min_ma" (default).
+#' @param max_ma \code{character}. The name of the column you wish to be treated
+#' as the maximum limit of the age range, e.g. "max_ma" (default).
+#' @param by \code{character}. How should the output be sorted?
+#' Either: "FAD" (first-appearance date; default), "LAD" (last-appearance data),
+#' or "name" (alphabetically by taxon names).
+#' @param plot \code{logical}. Should a plot of the ranges be generated?
+#'
+#' @return A \code{dataframe} containing the following columns:
+#' unique taxa (`taxa`), taxa ID (`taxon_id`), first appearance of taxon
```

Comment:
This is useful for plotting in this function rather than plotting by taxonomic names. I'd also argue it is easier to see differences between numbers than taxonomic names when they are similar!